### PR TITLE
(visualstudiocode) Migrate Visual Studio Code

### PR DESCRIPTION
**Pull request by pascalberger**
This PR is to move the Visual Studio Code package from https://github.com/jivkok/Chocolatey-AutomaticPackages/tree/master/VisualStudioCode to the Core Team Packages. Jivko and I would prefer to have the package in this repository for easier maintenance.

ID changes with this PR to all lower case as per [Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md). If this is a problem for Chocolatey I can change it back to the casing it had before.

If you're interested in accepting this PR I'll add Chocolatey as a maintainer to the package.

### CODE REVIEW

File: automatic/visualstudiocode/visualstudiocode.nuspec [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Features
+
+* **Meet IntelliSense:** Go beyond syntax highlighting and autocomplete with IntelliSense, which provides smart completions based on variable types, function definitions, and imported modules.
+* **Print statement debugging is a thing of the past:** Debug code right from the editor. Launch or attach to your running apps and debug with break points, call stacks, and an interactive console.
+* **Git commands built-in:** Working with Git has never been easier. Review diffs, stage files, and make commits right from the editor. Push and pull from any hosted Git service.
+* **Extensible and customizable:** Want even more features? Install extensions to add new languages, themes, debuggers, and to connect to additional services. Extensions run in separate processes, ensuring they won't slow down your editor.
+
+## Package parameters
+The package uses default install options except that it adds context menu entries and Visual Studio Code isn't started after installation.
```
Comment (majkinetor) [inline automatic/visualstudiocode/visualstudiocode.nuspec]: This should probably go into ## Notes

**Comment by majkinetor** [comment]
LGTM.

One note tho: chocolateyInstall could be substantially shorter. You can replace entire parameter section with few lines with the latest `chocolatey-core.extension`. See for example [virtualbox parmaeter handling](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/automatic/virtualbox/tools/chocolateyInstall.ps1#L12-L17) for example. It would be great to do so but it isn't mandatory (If you don't I will do it when its merged and provide optino to install extensions during installation).

File: automatic/visualstudiocode/visualstudiocode.nuspec [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+These parameters can be passed to the installer with the use of `-params`.
+For example: `-params '"/NoDesktopIcon /DontAddToPath"'`.
+
+## Notes
+The package uses default install options except that it adds context menu entries and Visual Studio Code isn't started after installation.
+    </description>
+    <tags>microsoft visualstudiocode vscode editor ide javascript typescript admin foss cross-platform</tags>
+    <dependencies>
+      <dependency id="DotNet4.5.2" />
```
Comment (AdmiringWorm) [inline automatic/visualstudiocode/visualstudiocode.nuspec]: why .NET 4.5.2? According to their [requirements page](https://code.visualstudio.com/docs/supporting/requirements)
it only needs .NET 4.5.

Comment (pascalberger) [inline automatic/visualstudiocode/visualstudiocode.nuspec]: This page is out of date. See [here](https://code.visualstudio.com/docs/setup/windows) which was [recently updated to 4.5.2](https://github.com/Microsoft/vscode-docs/pull/670)

Comment (pascalberger) [inline automatic/visualstudiocode/visualstudiocode.nuspec]: btw. I created a PR for fixing the docs: https://github.com/Microsoft/vscode-docs/pull/695

**Comment by AdmiringWorm** [comment]
Looks to be something wrong with the merge tasks logic,
since `desktopicon`,`quicklaunchicon` and `addtopath` is checked as the default in the installer,
those need to tested if they are false and add '!' before the task name

**Comment by majkinetor** [comment]
*To moderators*: I advise to squash this merge once it is ready as commit messages do not have `(visualstudiocode)` prefix.

**Comment by pascalberger** [comment]
@majkinetor I've updated the `chocolateyInstall.ps1` to use the parameter parsing from the `chocolatey-core.extension` extension.

@AdmiringWorm This should now be fixed together with the rewrite to the new parameter parsing.

I also prefixed all commits with `(visualstudiocode)`

**Comment by AdmiringWorm** [comment]
just had a quick look, and so far it looks good to me. but haven't been able to test it just yet.

Just a quick question, why the need to add the empty `VSCodeSetup-stable.exe.ignore` file?
I believe this is already handled since you used the `Install-ChocolateyPackage` helper, right?

**Comment by pascalberger** [comment]
@AdmiringWorm To be honest this file existed before I started maintaining it :) I assumed that it is necessary, but if this is already handled by `Install-ChocolateyPackage` I can remove it.

**Comment by majkinetor** [comment]
Thx, the difference is obvious, 2/3 of lines are removed. Few more can be removed if you actualy move this section:

```
+$packageName = 'visualstudiocode'
 +$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
 +$url32 = 'https://az764295.vo.msecnd.net/stable/7ba55c5860b152d999dda59393ca3ebeb1b5c85f/VSCodeSetup-1.7.2.exe'
 +$url64 = 'https://az764295.vo.msecnd.net/stable/7ba55c5860b152d999dda59393ca3ebeb1b5c85f/VSCodeSetup-1.7.2.exe'
 +$checksum32 = 'ce706a90880dfc94f1c6b3911ffd66be0ec4ec4c0fcb70eec78b20357bdcee23'
 +$checksum64 = 'ce706a90880dfc94f1c6b3911ffd66be0ec4ec4c0fcb70eec78b20357bdcee23'
```

into the `packageArgs` directly. This would require chanages on `au_SearchReplace` too.

Anyway, this can now be merged. I will certainly update this package.

**Comment by gep13** [comment]
@pascalberger @majkinetor @AdmiringWorm I have added the chocolatey user as a co-maintainer of this package on chocolatey.org.

@pascalberger can I assume that you and Jivko will no longer be pushing any package versions?  i.e. have you removed this package from any automatic processes you might have?

**Comment by pascalberger** [comment]
@gep13 I wrote @jivkok that he should remove the package from his repository and remove from his automatisation scripts.

**Comment by gep13** [comment]
@pascalberger do you know if he has done that?  Just want to avoid any race conditions for pushing a new package version.  This happened when we ported VirtualBox from @ferventcoder repository, and we don't want it to happen again.

**Comment by AdmiringWorm** [comment]
@gep13 looks like the `jobs.db` file on his repo was last updated 2 months ago and the visualstudiocode directory is stillthere.
So I think it is unlikely he have removed it yet.
I think we should wait with merging this until it's been removed from the old repo though

**Comment by pascalberger** [comment]
@gep13 @AdmiringWorm I'll let you know as soon as I get confirmation that Jivko has removed it from his side.

**Comment by majkinetor** [comment]
Why dont we follow migratiom guidelines here ?

**Comment by pascalberger** [comment]
@majkinetor I can make a PR for deleting the package from Jivkos repository, but I don't think he has the Ketarin stuff in the repo.

**Comment by gep13** [comment]
>@majkinetor said...
Why dont we follow migratiom guidelines here ?

I thought you didn't want to! 😛

**Comment by majkinetor** [comment]
I didn't want to do it for someone that doesn't reposnd. I generally think its a good idea.

**Comment by majkinetor** [comment]
> @majkinetor I can make a PR for deleting the package from Jivkos repository, but I don't think he has the Ketarin stuff in the repo.

> I didn't want to do it for someone that doesn't respond. I generally think its a good idea.

Let me take this back. I don't think this is good idea. Removing the ownership to user should be done always. User will not be able to take any action anyway and we can't clean entire world, its unreasonable.

**Comment by gep13** [comment]
>@majkinetor said...
Let me take this back. I don't think this is good idea. Removing the ownership to user should be done always. User will not be able to take any action anyway and we can't clean entire world, its unreasonable.

We have to be very careful when doing this, and I am hesitant to make this a rule.  

Removing a user, in my mind would reduce the possibility of them continuing to help with the package once it is moved to the core team repo, and this is something we really want to avoid.  Also, this is certainly NOT something we will be doing before reaching out to the current maintainers to ask the question.  People have went out of their way to create the packages in the first place, and simply put it would be bad form to remove them without consultation.

**Comment by majkinetor** [comment]
Yeah, you are right. 

Also, there is that script that dragon mentioned on a chat that I still have to look into.

**Comment by gep13** [comment]
@pascalberger any word from Jivko?

**Comment by pascalberger** [comment]
Not yet

Am 07.12.2016 8:49 vorm. schrieb "Gary Ewan Park" <notifications@github.com

**Comment by majkinetor** [comment]
OK, so clearly we have to set up tresholad on waiting - I propose 7 days for questions like this then we should remove the user as maintainer and send him a note.

**Comment by gep13** [comment]
>@majkinetor said...
I propose 7 days

I think I would be happier with 14 days.

**Comment by jivkok** [comment]
I have removed the VSCode from my Ketarin repo

**Comment by gep13** [comment]
>@jivkok said...
I have removed the VSCode from my Ketarin repo

That is great news!  Thank you for letting us know.  If yourself or @pascalberger have any other "high price" packages that you would like to move into the core team packages repo, and which you would still be happy to maintain here, please feel free to raise a PR.

**Comment by gep13** [comment]
@majkinetor remind me, what does the Push Required label mean again?

**Comment by majkinetor** [comment]
@gep13 : see [3.4 - Understanding labels](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#34-understanding-labels).

**Comment by gep13** [comment]
@majkinetor so in this instance, this PR is good to go, and we should simply wait for the next release of Visual Studio Code in order to fully test it out?  Or are we planning on forcing a package fix notation push?

**Comment by majkinetor** [comment]
I generally prefer fix when there are substantial change becuase any verification and after-push errors can be catched closer to the implementation. In this case I think we should skip it - VS updates frequently and I want to make aditional changes.

**Pull request merged**